python/sglang/srt/models/gemma2_reward.py

```diff
@@ -58,43 +58,11 @@ def forward(
         ), "Gemma2ForSequenceClassification is only used for embedding"
 
         hidden_states = self.model(input_ids, positions, forward_batch, input_embeds)
-        scores = self.score(hidden_states)
+        last_token_hidden = self.pooler(hidden_states, forward_batch).embeddings
+        scores = self.score(last_token_hidden)
 
-        return self.pooler(scores, forward_batch)
+        return EmbeddingPoolerOutput(scores)
 
-    def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
```

Comment:
can you also simplify the weight loader of `LlamaForSequenceClassification`?

Reply:
Done & verified.
![image](https://github.com/user-attachments/assets/2430e503-99a3-4277-87c9-47c340067362)